app.js

```diff
@@ -38,6 +57,95 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+app.delete('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+  const vegetableIdx = vegetables.findIndex(veggie => veggie.id == id)
+  if (vegetableIdx < 0) {
+    const message = `Could not find vegetable with ID of ${id}`
+    next({ status: 404, message })
+  }
+  const vegetable = vegetables.splice(vegetableIdx,1);
+  res.json(vegetable)
+})
+
+app.put('/vegetables/:id', helpers.validate, (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+  const vegetableIdx = vegetables.findIndex(vegetable => vegetable.id == id)
+  if (vegetableIdx < 0) {
+    const message = `Could not find vegetable with ID of ${id}`
+    next({ status: 404, message })
+  }
+  const vegetable = {id: req.params.id, name: req.body.name, price:req.body.price}
+  vegetables.splice(vegetableIdx,1,vegetable);
+  res.json(vegetable)
+})
+
+
+/**
+ * Fruit Routes
+ */
+
+app.get('/fruits', (req, res, next) => {
+  const { fruits } = data
+  if(req.query.name ) {
+    let matches = _.filter(fruits, (fruit) => {
+      return fruit.name.indexOf(req.query.name) > -1;
+    });
+    res.json(matches)
+  } else {
+    res.json(fruits)
+  }
+})
+
+app.get('/fruits/:id', (req, res, next) => {
+  const { fruits } = data
+  const { id } = req.params
+  const fruit = fruits.find(fruit => fruit.id === id)
+
+  if (!fruit) {
+    const message = `Could not find fruit with ID of ${id}`
+    next({ status: 404, message })
+  }
+
+  res.json(fruit)
+})
+
+app.post('/fruits', helpers.validate, (req, res, next) => {
+  const { fruits } = data
+  const fruit = { id: generateId(), ...req.body }
+
+  fruits.push(fruit)
+  res.status(201).json(fruit)
+})
+
+app.delete('/fruits/:id', (req, res, next) => {
+  const { fruits } = data
+  const { id } = req.params
+  const fruitIdx = fruits.findIndex(fruit => fruit.id == id)
+  if (fruitIdx < 0) {
+    const message = `Could not find fruit with ID of ${id}`
+    next({ status: 404, message })
+  }
+  
+  const fruit = fruits.splice(fruitIdx,1);
+  res.json(fruit)
```

Comment:
This will return an array as opposed to an object, which is what the instructions ask for.